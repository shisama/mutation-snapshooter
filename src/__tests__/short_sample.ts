import Vue from "vue";
// eslint-disable-next-line no-unused-vars
import Vuex, { ActionTree, MutationTree } from "vuex";
import snapshot from "../index";

Vue.use(Vuex);

const state = {
  counter: {
    value: 0
  }
};

type State = typeof state;

const mutations: MutationTree<State> = {
  increment: (state: State) => state.counter.value++
};

const actions: ActionTree<State, State> = {
  increment: ({ commit }) => commit("increment")
};
// eslint-disable-next-line no-unused-vars
const store = new Vuex.Store({
  state,
  mutations,
  actions
});

snapshot({
  store,
  dispatches: [dispatch => dispatch("increment")]
});
