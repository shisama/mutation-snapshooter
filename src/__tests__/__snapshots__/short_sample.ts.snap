// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`increment 1`] = `
"Snapshot Diff:
- Before
+ After

  Object {
    \\"counter\\": Object {
-     \\"value\\": 0,
+     \\"value\\": 1,
    },
  }"
`;
